### PR TITLE
update djax to use XLA dynamic shapes

### PR DESCRIPTION
**Pull request by mattjj**
Now that jaxlib==0.1.65 supports XLA dynamic shapes on CPU, we can avoid complexity in our own lowerings, at least for the cases that XLA dynamic shapes support (not involving DimIndexingExprs).

### CODE REVIEW

**Comment by machineko** [comment]
> Now that jaxlib==0.1.65 supports XLA dynamic shapes on CPU, we can avoid complexity in our own lowerings, at least for the cases that XLA dynamic shapes support (not involving DimIndexingExprs).

Hey Matthew is dynamic shapes on new XLA implementation have performance hit over fully static shapes? (I can't find information anywhere in docs 😄 and I can rember lax.dynamic_slice give me a hard performance drop few months ago should i give it a try again?)

**Comment by mattjj** [comment]
Closing this because we're going to land pieces in real JAX, rather than djax.

@machineko I didn't notice your comment until now, sorry! Yes, when the compiler isn't allowed to specialize its generated code on shape information we should expect a loss of performance, though exactly how much will depend on a lot of details (of the particular workload, of the backend, etc).
